### PR TITLE
Template deployment

### PR DESCRIPTION
**Pull request by mcardosos**
Thanks for your contribution! Please let us know what problems this PR addresses or what functionality it adds. Please link to other related issues and PRs to speed up review. Thank you!

### CODE REVIEW

**Comment by mcardosos** [comment]
@joshgav 
Fixes https://github.com/Azure-Samples/azure-sdk-for-go-samples/issues/108
Question, there were some templates for, apparently, deploying a webapp, I don't see them referenced anywhere in the repo, so I got rid of them. Are you using them somewhere else?
Another question, I have kept the original quickstart folder as it was before (plus some minor refactoring). It would be cool if we didn't duplicate code, however documentation depends on the original sample https://docs.microsoft.com/en-us/go/azure/azure-sdk-go-qs-vm . Are we planning on also updating the documentation so it reflects the migrated sample?

File: quickstarts/deploy-vm/main.go [hunk 1]
```diff
@@ -2,10 +2,9 @@ package main
 
 import (
 	"context"
-	"encoding/json"
-	"io/ioutil"
 	"log"
 
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/helpers"
```
Comment (joshgav) [inline quickstarts/deploy-vm/main.go]: I like the idea, but the quickstarts were intended to stand alone and not depend on other packages here. So I guess we should add the readJSON method back inline here.

File: resources/deployment.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	if err != nil {
+		return de, fmt.Errorf("cannot get the create deployment future respone: %v", err)
+	}
+
+	return future.Result(deployClient)
+}
+
+// ValidateDeployment validates the template deployments and their
+// parameters are correct and will produce a successful deployment.GetResource
+func ValidateDeployment(ctx context.Context, deploymentName string, template, params *map[string]interface{}) (valid resources.DeploymentValidateResult, err error) {
```
Comment (joshgav) [inline resources/deployment.go]: Nice! Thanks for adding this.

File: resources/resources.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	resourcesClient.Authorizer = autorest.NewBearerAuthorizer(token)
+	resourcesClient.AddToUserAgent(helpers.UserAgent())
+	return resourcesClient
+}
+
+// GetResource gets a resource, the generic way.
+// The API version parameter overrides the API version in
+// the SDK, this is needed because not all resources are
+// supported on all API versions.
+func GetResource(ctx context.Context, provider, resourceType, resourceName, apiVersion string) (resources.GenericResource, error) {
```
Comment (joshgav) [inline resources/resources.go]: This is nice. Can we rename `provider` to `resourceProvider` to match the next two parameter names? Thanks!

File: testdata/parameters.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (joshgav) [inline testdata/parameters.json]: Can we move this directory into the "resources" folder so it doesn't clutter up the root?

**Comment by joshgav** [comment]
@mcardosos 

> It would be cool if we didn't duplicate code, however documentation depends on the original sample 

The plan of record is that files under `quickstarts` are intended to stand alone and not depend on other parts of this repo even though code is duplicated. They also aren't tested in our CI system. So I think we need to put the `readJSON` method back in there and leave as is. If there are significant changes in that directory in the future we'll discuss with docs team first so they can update too. Thanks!

**Comment by mcardosos** [comment]
@joshgav what about the webapp templete?

File: testdata/parameters.json (hunk 4 above)
Comment (mcardosos) [reply testdata/parameters.json]: Fixed

File: resources/resources.go (hunk 3 above)
Comment (mcardosos) [reply resources/resources.go]: Fixed

File: quickstarts/deploy-vm/main.go (hunk 1 above)
Comment (mcardosos) [reply quickstarts/deploy-vm/main.go]: Fixed

**Pull request merged**